trio/_core/_asyncgens.py

```diff
@@ -12,6 +16,19 @@
 # Used to log exceptions in async generator finalizers
 ASYNCGEN_LOGGER = logging.getLogger("trio.async_generator_errors")
 
+if TYPE_CHECKING:
+    # TODO: Remove when dropping 3.8 support, __class_getitem__ added in 3.9
+    _WEAK_ASYNC_GEN_SET = weakref.WeakSet[AsyncGeneratorType[object, NoReturn]]
+    if sys.version_info < (3, 10):
```

Comment:
Since we're in a `TYPE_CHECKING` block, there isn't really any point checking the version - just use the builtin, type checkers know what it means. Mypy should suppress the version error, I think. If not just unconditionally import from typing.

Reply:
Changed in 07b26fa